### PR TITLE
test: Some fixes for cockpit-machines testcases

### PR DESCRIPTION
**Pull request by yunmingyang**
* Add some cases about storage pool
* 'testNetworkInfo' will not attach-interface
* Add some resource for run_test when using edge, otherwise send_keys() can not send correct string to the 'Operating System' input of creation dialog

### CODE REVIEW

File: test/selenium/selenium-machines-disks.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def testDetachDiskVmOff(self):
         self.click(self.wait_css('#vm-{}-disks-adddisk-dialog-add'.format(name), cond=clickable))
         self.wait_css('#vm-{}-disks-vda-device'.format(name))
         self.click(self.wait_css('#vm-{}-disks-vda-detach'.format(name), cond=clickable))
-        self.wait_css('vm-{}-disks-vda-device'.format(name), cond=invisible)
+        self.wait_css('#vm-{}-disks-vda-device'.format(name), cond=invisible)
```
Comment (KKoukiou) [inline test/selenium/selenium-machines-disks.py]: This is a fix unrelated to this commit and should be separated.

File: test/selenium/selenium-machines-storage-pool.py [hunk 2]
```diff
@@ -118,3 +124,131 @@ def testCreateNFSStoragePool(self):
...
+                                parttable='msdos' if part == 'dos' else part)
+            pool_name = self.create_storage_by_ui(name=name,
+                                                  storage_type='disk',
+                                                  target_path='/media',
+                                                  source_path=device,
+                                                  parted=part)
+            self.click(self.wait_css('#{}-name'.format(pool_name), cond=clickable))
+            self.click(self.wait_css('#delete-{}'.format(pool_name), cond=clickable))
+            self.click(
+                self.wait_xpath(
```
Comment (KKoukiou) [inline test/selenium/selenium-machines-storage-pool.py]: wait_css doesn't work here? Just out of curiosity.

File: test/selenium/testlib_avocado/machineslib.py [hunk 3]
```diff
@@ -201,7 +210,7 @@ def create_vm_by_ui(self,
                         name='default',
                         source_type='file',
                         source='/var/lib/libvirt/images/cirros.qcow2',
-                        operating_system='CirrOS',
+                        operating_system='CirrOS 0.4.0',
```
Comment (KKoukiou) [inline test/selenium/testlib_avocado/machineslib.py]: This change is also not related.

**Comment by KKoukiou** [comment]
@yunmingyang selenium failure looks related
```
testlib_avocado.exceptions.SeleniumElementFailure: Unable to locate name: #storage-pool-dialog-source > div > input
```

File: test/selenium/selenium-machines-storage-pool.py (hunk 2 above)
Comment (yunmingyang) [reply test/selenium/selenium-machines-storage-pool.py]: I think wait_css should work well. The reason why I use wait_xpath is a inspiration.If I use css_selector here, the path should be 'body > div:nth-child(2) > div.fade.in.modal > div > div > div.modal-footer > button.btn.btn-danger', then if I use xpath here, the path looks readable and clean.I think both of them is OK.Should I change wait_xpath to wait_css?

File: test/selenium/selenium-machines-disks.py (hunk 1 above)
Comment (yunmingyang) [reply test/selenium/selenium-machines-disks.py]: Separate it to another commit

File: test/selenium/testlib_avocado/machineslib.py (hunk 3 above)
Comment (yunmingyang) [reply test/selenium/testlib_avocado/machineslib.py]: Separate it to another commit

**Comment by KKoukiou** [comment]
@yunmingyang the selenium tests keep failing and the issues seem to be related to the changes.
Otherwise the code looks good, thanks.
In addition, can you add  'test' prifix to the commit message of https://github.com/cockpit-project/cockpit/pull/12691/commits/7580363ad13bccc9ae5eac5ec6c780255b061803

**Comment by yunmingyang** [comment]
OK, I am running locally and trying to resolve the issues. The prefix for [7580363](https://github.com/cockpit-project/cockpit/commit/7580363ad13bccc9ae5eac5ec6c780255b061803) will be also added very soon.Many thanks

**Comment by KKoukiou** [comment]
@yunmingyang still failures on edge https://209.132.184.41:8493/logs/pull-12691-20190910-081717-aa827c65-cockpit-project-cockpit--fedora-30-selenium-edge/log.html#42

**Comment by yunmingyang** [comment]
@KKoukiou I check the screenshot of failed cases, they all failed with an Oops on the page, selenium can not locate the element， thus the cases failed.Then, I check some logs of failed cases, it seems that they all failed in the step which is switch to the frame of machines, so maybe the Oops is raised when switch to the machines frame.I have no idea about the reason why cases failed, because I didn't change the code in the part which is to switch to the machines frame.And, actually, I have passed the edge test with 'run-test' locally. So should we just re-trigger the edge test or get the reason about the Oops?

**Comment by KKoukiou** [comment]
> @KKoukiou I check the screenshot of failed cases, they all failed with an Oops on the page, selenium can not locate the element， thus the cases failed.Then, I check some logs of failed cases, it seems that they all failed in the step which is switch to the frame of machines, so maybe the Oops is raised when switch to the machines frame.I have no idea about the reason why cases failed, because I didn't change the code in the part which is to switch to the machines frame.And, actually, I have passed the edge test with 'run-test' locally. So should we just re-trigger the edge test or get the reason about the Oops?

It you can give me the console logs from the Edge browser it would be great. But it should not really block this PR. Waiting for the rest of the tests to run.

**Comment by martinpitt** [comment]
> selenium can not locate the element

Just in case you are not aware: Edge's selenium driver has an utterly annoying bug that it does not "see" elements of the DOM which are not visible on the screen (this works fine with firefox and chromium). Unfortunately the windows-10 VM does not run with a high resolution, so the window is quite small (something like 1024x768?). Possibly this needs some more [scrollIntoView hacks like that](https://github.com/cockpit-project/cockpit/commit/b960b0fc7b)?

**Comment by yunmingyang** [comment]
@martinpitt, many thanks. But maybe the issue is not related with the scroll bar, because the page is white screen when switch to the 'machines' page and there are also an Oops. Elements do not appear on the page, so the cases of machines are failed.I have no idea about the reason of white screen. Emmm....I add another 1G RAM and 1 CPU for the edge test, does it relate with the failed cases?

**Comment by KKoukiou** [comment]
@martinpitt are you ok if I unblock these - since the edge failure is not coming from this PR..

**Comment by martinpitt** [comment]
@KKoukiou sure! But please let's squash this at least a little bit.

**Comment by KKoukiou** [comment]
> @KKoukiou sure! But please let's squash this at least a little bit.

rebased and added closes - but squashing does not fit here I think - the commits are mostly tiny standone changes.

File: test/selenium/run-tests [hunk 4]
```diff
@@ -331,7 +331,7 @@ class SeleniumWinTestSuite(SeleniumTestSuite):
     def _prepare_selenium(self):
         if not self.machine_selenium:
             self.machine_avocado.dhcp_server(range=[self.ipaddr_selenium, self.ipaddr_selenium])
-            self.machine_selenium = testvm.VirtMachine(image='windows-10', verbose=self.verbose, networking=self.network.host())
+            self.machine_selenium = testvm.VirtMachine(image='windows-10', verbose=self.verbose, networking=self.network.host(), memory_mb=2048, cpus=2)
```
Comment (martinpitt) [inline test/selenium/run-tests]: Ugh, good catch! No wonder this machine feels like a pit of tar when debugging interactively.

**Comment by martinpitt** [comment]
Still [some](https://logs-https-cockpit.apps.ci.centos.org/logs/pull-12691-20190924-083131-7d8aef0c-cockpit-project-cockpit--fedora-30-selenium-firefox/log.html) [failures](https://209.132.184.41:8493/logs/pull-12691-20190924-083131-7d8aef0c-cockpit-project-cockpit--fedora-30-selenium-chrome/log.html), retrying.
